src/coreclr/vm/methodtable.inl

```diff
@@ -415,6 +415,10 @@ inline MethodDesc* MethodTable::GetMethodDescForSlot(DWORD slot)
     CONTRACTL_END;
 
     PCODE pCode = GetRestoredSlot(slot);
+    if (pCode == (PCODE)NULL)
```

Comment:
This does not look like the right fix. I think that the problem is in the caller. The caller should call some other method. @davidwrighton ?

Reply:
It looks like I should change the caller (ExceptionTypeOverridesStackTraceGetter) to use GetMethodDescForSlot_NoThrow() and remove this change.  @davidwrighton?

Reply:
Yeah, that looks like a good change. I'm taking a look at removing `GetMethodDescForSlot`/`GetRestoredSlot` from the DAC build entirely to see what else might need to be fixed.